modules/dcache/src/main/java/org/dcache/pool/statistics/IoStatistics.java

```diff
@@ -42,20 +46,25 @@
      *         The duration of the read request in nanoseconds; must be non-negative
      */
 
-    public void updateRead(long readBytes, long readTime) {
+    public void updateRead(long readBytes, long readTime, long requestedBytes) {
 
-        double readSpeed = calculateSpeed(readBytes, readTime);
-        _readSpeeds[_readSpeedsArrayIndex] = readSpeed;
-        _readSpeedsArrayIndex = (_readSpeedsArrayIndex + 1) % _readSpeeds.length;
+        if(readBytes >= 0) {
+            double readSpeed = calculateSpeed(readBytes, readTime);
+            _readSpeeds[_readSpeedsArrayIndex] = readSpeed;
+            _readSpeedsArrayIndex = (_readSpeedsArrayIndex + 1) % _readSpeeds.length;
 
-        _minReadSpeed = (_readRequestNum == 0) ? readSpeed : Math.min(readSpeed, _minReadSpeed);
-        _maxReadSpeed = (_readRequestNum == 0) ? readSpeed : Math.max(readSpeed, _maxReadSpeed);
-        _avgReadSpeed = (_readRequestNum == 0) ? readSpeed : calculateNewAvg(_readRequestNum,_avgReadSpeed, readSpeed);
-        _95ReadSpeed = (_readRequestNum == 0) ? readSpeed : percentile.evaluate(_readSpeeds, 0, (_readSpeedsArrayIndex - 1));
+            _minReadSpeed = (_readRequestNum == 0) ? readSpeed : Math.min(readSpeed, _minReadSpeed);
```

Comment:
These are all good points and I fixed it! Thank for your patience :)

The only problem I see in initializing _minReadSpeed with Double.MAX_VALUE and _maxReadSpeed with Double.MIN_VALUE is that when the methods updateRead and updateWrite are never called but the statistics are however evaluated the values of these to parameters could be confusing or are they not?